TeamCity: Refactor config tests

SarahFrench opened the pull request:
<!-- Put a description of what this PR is for here, along with any references to issues that this resolves or contributes to -->


<!--
Please self-review your PR against the review checklist before creating it: https://googlecloudplatform.github.io/magic-modules/contribute/review-pr/

Completing the checklist will help speed up the review process, and we appreciate you spending time on them before sending
your code to be reviewed.

If your PR is still work in progress, please create it in draft mode
-->

<!-- AUTOCHANGELOG for Downstream PRs.

Please select one of the following "release-note:" headings:
    - release-note:enhancement
    - release-note:bug
    - release-note:note
    - release-note:new-resource
    - release-note:new-datasource
    - release-note:deprecation
    - release-note:breaking-change
    - release-note:none

Unless you choose release-note:none, please add a release note.

See https://googlecloudplatform.github.io/magic-modules/contribute/release-notes/ for writing good release notes.

You can add more release note blocks if you want more than one CHANGELOG
entry for this PR.
-->
**Release Note Template for Downstream PRs (will be copied)**

```release-note:none

```

SarahFrench commented:
I'm going to keep this as draft until the GHA implemented in https://github.com/GoogleCloudPlatform/magic-modules/pull/9954 is available to help reviewers

modular-magician commented:
Hi there, I'm the Modular magician. I've detected the following information about your changes:

## Diff report
Your PR generated some diffs in downstreams - here they are.

Terraform GA: [Diff](https://github.com/modular-magician/terraform-provider-google/compare/auto-pr-9956-old..auto-pr-9956) ( 3 files changed, 42 insertions(+), 61 deletions(-))

SarahFrench commented:
Rebased to include https://github.com/GoogleCloudPlatform/magic-modules/pull/9954

SarahFrench commented on .github/workflows/teamcity-pr-checks.yml:
```diff
@@ -6,7 +6,7 @@ on:
     paths:
       - '.github/workflows/teamcity-pr-checks.yml'
-      - 'mmv1/third_party/terraform/!.teamcity'
+      - 'mmv1/third_party/terraform/.teamcity'
```
I thought that the `!` was required to use a folder starting with a dot but looks like I was wrong!

modular-magician commented:
Hi there, I'm the Modular magician. I've detected the following information about your changes:

## Diff report
Your PR generated some diffs in downstreams - here they are.

Terraform GA: [Diff](https://github.com/modular-magician/terraform-provider-google/compare/auto-pr-9956-old..auto-pr-9956) ( 3 files changed, 44 insertions(+), 63 deletions(-))

SarahFrench commented:
I'm going to wait for https://github.com/GoogleCloudPlatform/magic-modules/pull/10003 to be merged and rebase again 🤦

SarahFrench commented:
Yay, after the last rebase the TeamCity config check is running as expected 🎉

modular-magician commented:
Hi there, I'm the Modular magician. I've detected the following information about your changes:

## Diff report
Your PR generated some diffs in downstreams - here they are.

Terraform GA: [Diff](https://github.com/modular-magician/terraform-provider-google/compare/auto-pr-9956-old..auto-pr-9956) ( 3 files changed, 44 insertions(+), 63 deletions(-))

SarahFrench commented:
Rebased to include https://github.com/GoogleCloudPlatform/magic-modules/pull/10025, addressed merge conflicts and used this PR's work to refactor the new tests added

modular-magician commented:
Hi there, I'm the Modular magician. I've detected the following information about your changes:

## Diff report
Your PR generated some diffs in downstreams - here they are.

Terraform GA: [Diff](https://github.com/modular-magician/terraform-provider-google/compare/auto-pr-9956-old..auto-pr-9956) ( 3 files changed, 48 insertions(+), 80 deletions(-))
